resources/web/wwi/Animation.js

```diff
@@ -0,0 +1,763 @@
+    document.getElementById('gtao-pane').style.visibility = 'hidden';
+    document.getElementById('gtao-display').innerHTML = event.srcElement.id;
+    document.getElementById('settings-pane').style.visibility = 'visible';
+    for (let i of document.getElementsByClassName('check-gtao')) {
+      if (i.id === 'c' + event.srcElement.id)
+        i.innerHTML = '&check;';
+      else
+        i.innerHTML = '';
+    }
+    this.start = new Date().getTime() - this.data.basicTimeStep * this.step / this.speed;
+    this.scene.render();
+  }
+
+  _openGtaoPane() {
+    document.getElementById('settings-pane').style.visibility = 'hidden';
+    document.getElementById('gtao-pane').style.visibility = 'visible';
+  }
+
+  _closeGtaoPane() {
+    document.getElementById('settings-pane').style.visibility = 'visible';
+    document.getElementById('gtao-pane').style.visibility = 'hidden';
+  }
+
+  _gtaoLevelToText(number) {
+    let string = '';
+    switch (number) {
+      case 1:
+        string = 'Low';
+        break;
+      case 2:
+        string = 'Medium';
+        break;
+      case 3:
+        string = 'High';
+        break;
+      case 4:
+        string = 'Ultra';
+        break;
+    }
+
+    return string;
+  }
+
+  textToGtaoLevel(text) {
+    let level = 4;
+    switch (text) {
+      case 'Low':
+        level = 1;
+        break;
+      case 'Medium':
+        level = 2;
+        break;
+      case 'High':
+        level = 3;
+        break;
+      case 'Ultra':
+        level = 4;
+        break;
+    }
+
+    return level;
+  }
+
+  _createPlayBar() {
+    const div = document.createElement('div');
+    div.id = 'play-bar';
+    this.view.view3D.appendChild(div);
+
+    div.addEventListener('mouseover', () => this._showPlayBar());
+    div.addEventListener('mouseleave', _ => this._onMouseLeave(_));
+
+    let leftPane = document.createElement('div');
+    leftPane.className = 'left-pane';
+    leftPane.id = 'left-pane';
+
+    let rightPane = document.createElement('div');
+    rightPane.className = 'right-pane';
+    rightPane.id = 'right-pane';
+
+    document.getElementById('play-bar').appendChild(leftPane);
+    document.getElementById('play-bar').appendChild(rightPane);
+    this.view.mouseEvents.hidePlayBar = this._hidePlayBar;
+    this.view.mouseEvents.showPlayBar = this._showPlayBar;
+  }
+
+  _createSlider() {
+    window.customElements.define('my-slider', AnimationSlider);
+    let timeSlider = document.createElement('my-slider');
+    timeSlider.id = 'time-slider';
+    let those = this;
+    document.addEventListener('slider_input', (e) => { those._updateSlider(e); });
+    document.getElementById('play-bar').appendChild(timeSlider);
+    document.querySelector('my-slider').shadowRoot.getElementById('range').addEventListener('mousemove', _ => this._updateFloatingTimePosition(_));
+    document.querySelector('my-slider').shadowRoot.getElementById('range').addEventListener('mouseleave', _ => this._hideFloatingTimePosition(_));
```

Comment:
I am not sure it is a good idea in this case, because:

- We want to create the toolbar once the world is ready and the .json loaded and not before.
- It would result in a lot of DOM elements in which we are not really interested to have them as attributes because we use them only once